### DISCUSSION TITLE
getting a subset of node ids from tables

### DISCUSSION THREAD

**Discussion by alxsimon** [tskit-dev/tskit#2289] [Q&A] [answered]
Hi everyone,

maybe a silly question here, but is there a more efficient way of getting the node ids of some subset from the tables than doing this?
```
nodes = ts.tables.nodes
numpy.where(nodes.population == 1)[0]
```
Which I find a bit cumbersome to write and often forget taking the first element at the end.

I get the following is not possible as there is no stable id column:
```
nodes.id[nodes.population == 1]
or 
nodes[nodes.population == 1].id
```
But why? Why are ids changing when subsetting, and not a "real" column?
It's not really prettier than what's written above but seems more straightforward to me (surely remnants from my R days).

Thanks a lot
Alexis

**Comment by benjeffery** [comment, marked as answer]
Hi @alxsimon, thanks for the question.
The `id` of a row in tskit is just its index in the table, so they will change when you take a subset of the table. This saves storing an extra field in each table and also saves thorny id collision issues when manipulating tables. IDs aren't needed as you can use techniques such as your first example.

**Reply by alxsimon** [reply]
Thanks for the answer.

**Comment by hyanwong** [comment]
You can always use `np.flatnonzero(condition)` instead of `np.where(condition)[0]` if you don't like appending `[0]`. But it sounds kinda ugly. I agree, it's not terribly intuitive.
